### ISSUE TITLE
miracast base does not work

### ISSUE THREAD

**Issue by lucapizz** [albfan/miraclecast#363]
Hi,
I just tryied to install the image on my RPi 3 without wifi. Then also on a Rpi3 with integrated wifi.
I followed this:
https://github.com/albfan/miraclecast/wiki/raspberry-pi-3

when on the terminal I send the command
sudo miracle-wifid
I get no answer, so if I type CTRL+C I get the message
Caught signal 2, exiting…
ERROR: dbus:syscall filed (-107): Transport endpoint is not connected

where I failed the process?
Any help would be appreciated
Thank you

**Comment by albfan** [comment]
Check the FAQ about logs useful to get support. https://github.com/albfan/miraclecast/wiki/FAQ#log-for-support

**Comment by lucapizz** [comment]
[lucalog.zip](https://github.com/albfan/miraclecast/files/4019217/lucalog.zip)
Hi albfan,
her is my log.

unfortunately when I type the command miracle-wifid I can get any log and any answer
so I typed ctlr+c to exit after a while

I flashed the image on my sdcard with win32diskimager on Windows.
My sdcard is 8 Gbytes and I think it is enough

Hoping this can help to understand the problem

bye
luca

**Comment by albfan** [comment]
You need to run it as:

    sudo miracle-wifid --log-level trace

**Comment by lucapizz** [comment]
[luca2log.zip](https://github.com/albfan/miraclecast/files/4020705/luca2log.zip)
Hi,
I typed again the commands and this time got the wifid.log 
in the term2copy.txt I stored the copy of what I see on the terminal.

what do you think about it? 
I do not understand much.
bye and thank you a lot

**Comment by lucapizz** [comment]
Just to clarify my steps:
- I wrote the image on my 8Gb sdcard
- turned on my RASPI  (no wifi integrated)
- I skipped the modify of config.txt because I am not on pi 1 nor 2
- I skipped the yaourt command because I think it is already made in the image.

The miracle does not work

so I tryed to do the yaourt step and after all the process the miracle do something but with trouble that push me to asking help.

I wrote this steps to ask, may be this sequence the problem source?
I don't know but just in case it can help…

thank you again

PS: I re tried the same on another pi 3 with integrated wifi, but with the same results.

**Comment by lucapizz** [comment]
I have another question/clue

If I try to launch Applications/Internet/Firefox the mouse arrow turns to a rotating icon and after a while it come back to arrow but the firefox browser does not start.

is it normal?
this is in such way from the first run of the image and never changed.


bye

**Comment by albfan** [comment]
Everything seems to work as expected. See you need to keep `miracle-wifid` running (is a daemon) while run `miracle-sinkctl` Seems you stopped it, that's why you see

> ERROR: cannot retrieve objects: The name org.freedesktop.miracle.wifi was not provided by any .service files

as `miracle-wifid` is the one that provide those services.

**Comment by lucapizz** [comment]
Hi,
I do not understand.
when I type the command "sudo miracle-wifid"
is it normal do not get any answer?

however, to leave it runnning, now I opened onother terminal and tyiping the
command "sudo miracle-sinkctl" I get a new line light blue like this
[miraclectl] # 

now I opened a third terminal and typing ps -ef I see two interesting lines following
1) root . . .. .pts/1   00:00:00 sudo miracle-wifid --log-level t
2) root .....   pts/0   00:00:00 sudo miracle-sinkctl

but I do not know what it means.

now I expect that trying on my mobile (android) to activate "MirrorShare" I will find a device to connect to but nothing…

So could you explain what I have to do ?
May be I have not understood nothing but I would like to connect my mobile to raspi and see it on the TV of the raspi.

ah, a last important thing, typing "ping 8.8.8.8" on terminal I get always "connect: Network is unreachable"


Thank you for your help
bye

**Comment by albfan** [comment]
Yes it's normal there's no output, it will be only if you request with --log-level. A daemon is there to be used with other tools. You need it to run in background (or in a different terminal as you did)

Now you need to selected the interface on miracle-sinkctl (normally "run 3")

all that is described in README

**Comment by lucapizz** [comment]
Hi albfan,
thanks for your support. Now I am starting to understand a little more.
I tried to use the raspi without wifi and I have seen a miracast on my lan, but after a first connection I saw an error and I no more able to re do the same.

question: Can I use miracast throug cable lan? I think yes but just to be sure, can I?

now I try to re do yaourt and restart from the beginning

question: May I follow just the readme instruction, after yaourt install and reboot?

tks & bye

**Comment by lucapizz** [comment]
Hi,
I have one good news and some bad news

After re do the yaourt on the raspi 3 with no wifi I get a connection via miracast through cable lan.

I have seen a video on the TV connected to my raspi. wonderful!

The bad news.
No audio on the TV.
The video is a little blocking, like if it skip some images. Not fluent.
The speed through the cable lan I think is better than wifi, do you agree?
At the end, once I disconnect graceful from my mobile phone I have seen some error on raspi terminal and I can re connect.

So I am a little unsatisfied, I don't know if the guilty is the raspi only or if the android make his part but a question: If I buy a dongle (Chromecast like) I will get the same results?

Do you have better results? if yes I am wrong in somewhat, but what?

Should I try to install raspbian and then put on it miracast in some way?

thank you for your support
bye

**Comment by lucapizz** [comment]
At the end, once I disconnect graceful from my mobile phone I have seen some error on raspi terminal and I can NOT re connect.

**Comment by albfan** [comment]
Not sure what you mean with no WiFi. Miraclecast is an implementation of Miracast which works through WiFi direct (there's no need for WiFi network, but streaming happens through WiFi interface)

**Comment by lucapizz** [comment]
I mean that I hav a version of raspi with integrated wifi hardware and another without, may be raspi 3 and raspi 3+b I don't remember

**Comment by lucapizz** [comment]
Hi,
now I realized that I am using Raspi Pi Model 2B.

So I will retry all from the beginning.

Now if you agree I will close this issue and eventually will re open a new one if I face more trouble.

What do you think?

**Comment by lucapizz** [comment]
And 
when I said the raspi with and without wifi, I was thinking to work on a raspi 3 (with wifi integrated) while I was really using a raspi 2 (without wifi integrated)
bye
